providers/intercom/search.go

```diff
@@ -0,0 +1,116 @@
+package intercom
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+// Search is used when `Since` parameter is provided for the select few objects.
+// Documentation below explains how search is done for "conversations" object. Others, use the same query language.
+// https://developers.intercom.com/docs/references/rest-api/api.intercom.io/conversations/searchconversations
+// https://developers.intercom.com/docs/references/rest-api/api.intercom.io/contacts/searchcontacts
+func (c *Connector) readViaSearch(
+	ctx context.Context, config common.ReadParams,
+) (*common.JSONHTTPResponse, *urlbuilder.URL, error, bool) {
+	if !incrementalSearchObjectPagination.Has(config.ObjectName) {
+		return nil, nil, nil, false
+	}
+
+	if config.Since.IsZero() && config.ObjectName != ticketsObjectName {
+		// Search is only relevant when we do incremental reading.
+		// Tickets is an exception. We can do full read only via POST.
+		return nil, nil, nil, false
+	}
+
+	url, err := constructURL(c.BaseURL, config.ObjectName, "search")
+	if err != nil {
+		return nil, nil, err, true
+	}
+
+	conversation, err := c.createSearchPayload(config)
+	if err != nil {
+		return nil, nil, err, true
+	}
+
+	rsp, err := c.Client.Post(ctx, url.String(), &conversation, apiVersionHeader)
+	if err != nil {
+		return nil, nil, err, true
+	}
+
+	return rsp, url, nil, true
+}
+
+func (c *Connector) createSearchPayload(params common.ReadParams) (*searchReqPayload, error) {
+	if params.Since.IsZero() && params.ObjectName == ticketsObjectName {
+		// Perform full read for tickets using POST query.
+		// This is a hack, query is designed to return all objects.
+		return &searchReqPayload{
+			Query: searchQuery{
```

Comment:
@Cobalt0s this can also be done for a couple more objects, right? Off the top of my head, articles would be one of them

Reply:
Articles search is done via GET and has no time parameter to benefit `Since` filter.

Docs: https://developers.intercom.com/docs/references/rest-api/api.intercom.io/articles/searcharticles